lib/request.js

```diff
@@ -104,7 +104,7 @@ class Request extends AsyncResource {
       }
 
       if (!hostHeader) {
-        header += `host: ${hostname}\r\n`
+        header += `host: ${hostname || 'localhost'}\r\n`
```

Comment:
This should never happen? hostname should always have a value?

Reply:
I see `unix:///`

Reply:
Is `localhost` correct here?

Reply:
I was hitting this case -- believe if no host name is passed this value is initialized as `''`. I didn't validate we were only hitting it through the UDS path, but it seemed likely the TCP path hits this as well.

I only noticed this because my server was returning a 400 complaining `Host` hadn't been set.